extensions/opentelemetry/opentelemetry/runtime/src/main/java/io/quarkus/opentelemetry/runtime/tracing/vertx/HttpInstrumenterVertxTracer.java

```diff
@@ -0,0 +1,459 @@
+        @Override
+        public Iterable<String> keys(final HttpRequest carrier) {
+            return carrier.headers().names();
+        }
+
+        @Override
+        public String get(final HttpRequest carrier, final String key) {
+            if (carrier == null) {
+                return null;
+            }
+
+            return carrier.headers().get(key);
+        }
+    }
+
+    private static class ClientAttributesExtractor extends HttpClientAttributesExtractor<HttpRequest, HttpResponse> {
+        @Override
+        protected String url(final HttpRequest request) {
+            return request.absoluteURI();
+        }
+
+        @Override
+        protected String flavor(final HttpRequest request, final HttpResponse response) {
+            return null;
+        }
+
+        @Override
+        protected String method(final HttpRequest request) {
+            return request.method().name();
+        }
+
+        @Override
+        protected List<String> requestHeader(final HttpRequest request, final String name) {
+            return request.headers().getAll(name);
+        }
+
+        @Override
+        protected Long requestContentLength(final HttpRequest request, final HttpResponse response) {
+            return null;
+        }
+
+        @Override
+        protected Long requestContentLengthUncompressed(final HttpRequest request, final HttpResponse response) {
+            return null;
+        }
+
+        @Override
+        protected Integer statusCode(final HttpRequest request, final HttpResponse response) {
+            return response.statusCode();
+        }
+
+        @Override
+        protected Long responseContentLength(final HttpRequest request, final HttpResponse response) {
+            return null;
+        }
+
+        @Override
+        protected Long responseContentLengthUncompressed(
+                final HttpRequest request, final HttpResponse response) {
+            return null;
+        }
+
+        @Override
+        protected List<String> responseHeader(final HttpRequest request, final HttpResponse response, final String name) {
+            return response.headers().getAll(name);
+        }
+    }
+
+    private static class HttpRequestTextMapSetter implements TextMapSetter<HttpRequest> {
+        @Override
+        public void set(final HttpRequest carrier, final String key, final String value) {
+            if (carrier != null) {
+                carrier.headers().set(key, value);
+            }
+        }
+    }
+
+    static class HttpRequestSpan implements HttpRequest {
+        private final HttpRequest httpRequest;
+        private final MultiMap headers;
+        private final Context context;
+        private final io.opentelemetry.context.Context spanContext;
+
+        HttpRequestSpan(
+                final HttpRequest httpRequest,
+                final MultiMap headers,
+                final Context context,
```

Comment:
From this case, I believe you require this context to be a duplicated context. 
It would be safer to check that you get what you expect.

Reply:
You may need to create your own duplicated context if you get a root context.

Reply:
Here is a code snipet to capture a duplicated context, and create if needed:
```java
 private Context captureTheRightContext(Vertx vertx) {
        Context context = Vertx.currentContext();
        if (context == null) {
            // Create both a root and a duplicated context
            context = ((ContextInternal) vertx.getOrCreateContext()).duplicate();
        } else {
            if (isRootContext(context)) {
                context = ((ContextInternal) context).duplicate();
            }
        }
        return context;
    }

    private boolean isRootContext(Context context) {
        return context instanceof EventLoopContext || context instanceof WorkerContext;
    }
```